core/src/apps/binance/sign_tx.py

```diff
@@ -0,0 +1,65 @@
+from trezor.crypto.curve import secp256k1
+from trezor.crypto.hashlib import sha256
+from trezor.messages import MessageType
+from trezor.messages.BinanceSignedTx import BinanceSignedTx
+from trezor.messages.BinanceTxRequest import BinanceTxRequest
+
+from apps.binance import CURVE, helpers, layout
+from apps.common import paths
+
+
+async def sign_tx(ctx, envelope, keychain):
+    # create transaction message -> sign it -> create signature/pubkey message -> serialize all
+    await paths.validate_path(
+        ctx, helpers.validate_full_path, keychain, envelope.address_n, CURVE
+    )
+
+    node = keychain.derive(envelope.address_n)
+
+    tx_req = BinanceTxRequest()
+
+    msg = await ctx.call(
+        tx_req,
+        MessageType.BinanceCancelMsg,
+        MessageType.BinanceOrderMsg,
+        MessageType.BinanceTransferMsg,
+    )
+
+    msg_json = helpers.produce_json_for_signing(envelope, msg)
+    signature_bytes = generate_content_signature(msg_json.encode(), node.private_key())
+
+    if msg.MESSAGE_WIRE_TYPE == MessageType.BinanceTransferMsg:
```

Comment:
each of the branches should be a single function in `layout.py` taking `msg` as the argument.

also in cases you have multiple items per branch (inputs and outputs here, three features of the order), these should be a single paginated layout instead of sequence of confirm screens

Reply:
I changed the layouts as requested. is one signature = one paginated confirmation the preferred ui pattern?